explorer/explorer_pods.go

```diff
@@ -4,33 +4,62 @@ import (
 	"fmt"
 	"strings"
 
+	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
 )
 
 const (
 	podLabel = "Pod"
 )
 
 type PodsExplorer struct {
-	Items              []*MenuItem
-	PreviousItem       *MenuItem
-	NamespaceToExplore string
-	Filters            map[string]string
-	PreviousExplorer   Explorable
+	Items                []*MenuItem
+	PreviousItem         *MenuItem
+	NamespaceToExplore   string
+	Filters              map[string]string
+	PreviousExplorer     Explorable
+	PreviousResourceName string
 }
 
 func (n *PodsExplorer) List() error {
 	n.Items = []*MenuItem{}
 
-	// Pods
-	pods, err := k8sclient.CoreV1().Pods(n.NamespaceToExplore).List(v1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(n.Filters).String(),
-	})
+	// List Pods
+	pods, err := k8sclient.Core().Pods(n.NamespaceToExplore).List(v1.ListOptions{})
 	if err != nil {
 		return err
 	}
-	for _, item := range pods.Items {
+
+	var podlist []apiv1.Pod
+	switch n.PreviousExplorer.Kind() {
+	case "deployment":
+		// get replicasets
+		replicaSetMap := make(map[string]bool)
+		replicaSets, err := k8sclient.Extensions().ReplicaSets(n.NamespaceToExplore).List(v1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, replicaset := range replicaSets.Items {
+			for _, owner := range replicaset.GetOwnerReferences() {
+				if strings.ToLower(owner.Name) == n.PreviousResourceName {
+					replicaSetMap[replicaset.GetName()] = true
+					break
+				}
+			}
+		}
+
+		// get pods
+		podlist = getOwnerPods(pods.Items, replicaSetMap, "replicaset")
+
+	case "statefulset":
```

Comment:
if we are going to use strings to map all this can we pull them into a constant instead of hard coding?